## Read Data

In [ ]:
# Importing required libraries and fixing options
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

pd.options.display.max_columns = None
pd.options.display.max_rows = None

%matplotlib inline

In [ ]:
# Import data from URL

url = "https://drive.google.com/file/d/1HhOnZXyXkagtgd5TfzM61HPRuuykJ5AN/view?usp=sharing"
url = 'https://drive.google.com/uc?id=' + url.split('/')[-2]
hnp_df = pd.read_csv(url)
hnp_df.head()

Country Name Country Code  \
0  Africa Eastern and Southern          AFE   
1  Africa Eastern and Southern          AFE   
2  Africa Eastern and Southern          AFE   
3  Africa Eastern and Southern          AFE   
4  Africa Eastern and Southern          AFE   

                                      Indicator Name  Indicator Code  \
0  Adolescent fertility rate (births per 1,000 wo...     SP.ADO.TFRT   
1  Adults (ages 15+) and children (0-14 years) li...     SH.HIV.TOTL   
2  Adults (ages 15+) and children (ages 0-14) new...  SH.HIV.INCD.TL   
3                  Adults (ages 15+) living with HIV     SH.DYN.AIDS   
4        Adults (ages 15-49) newly infected with HIV     SH.HIV.INCD   

         1960        1961        1962       1963       1964        1965  \
0  140.120535  140.594564  141.128109  141.32344  141.50816  141.668896   
1         NaN         NaN         NaN        NaN        NaN         NaN   
2         NaN         NaN         NaN        NaN        NaN         NaN   
3         NaN         NaN         NaN        NaN        NaN         NaN   
4         NaN         NaN         NaN        NaN        NaN         NaN   

         1966        1967        1968        1969        1970        1971  \
0  141.879704  141.983225  142.020168  142.010506  142.033458  142.279778   
1         NaN         NaN         NaN         NaN         NaN         NaN   
2         NaN         NaN         NaN         NaN         NaN         NaN   
3         NaN         NaN         NaN         NaN         NaN         NaN   
4         NaN         NaN         NaN         NaN         NaN         NaN   

         1972        1973        1974        1975        1976        1977  \
0  142.629356  142.469778  142.290486  142.053901  141.730609  141.396423   
1         NaN         NaN         NaN         NaN         NaN         NaN   
2         NaN         NaN         NaN         NaN         NaN         NaN   
3         NaN         NaN         NaN         NaN         NaN         NaN   
4         NaN         NaN         NaN         NaN         NaN         NaN   

         1978        1979        1980        1981        1982        1983  \
0  140.774087  140.150094  139.513693  138.804554  138.071244  137.058143   
1         NaN         NaN         NaN         NaN         NaN         NaN   
2         NaN         NaN         NaN         NaN         NaN         NaN   
3         NaN         NaN         NaN         NaN         NaN         NaN   
4         NaN         NaN         NaN         NaN         NaN         NaN   

         1984       1985        1986        1987        1988        1989  \
0  136.059339  135.08785  134.103381  133.152267  132.012111  130.869067   
1         NaN        NaN         NaN         NaN         NaN         NaN   
2         NaN        NaN         NaN         NaN         NaN         NaN   
3         NaN        NaN         NaN         NaN         NaN         NaN   
4         NaN        NaN         NaN         NaN         NaN         NaN   

         1990        1991        1992        1993        1994        1995  \
0  129.717498  128.562197  127.400373  126.969852  126.495358  125.972751   
1         NaN         NaN         NaN         NaN         NaN         NaN   
2         NaN         NaN         NaN         NaN         NaN         NaN   
3         NaN         NaN         NaN         NaN         NaN         NaN   
4         NaN         NaN         NaN         NaN         NaN         NaN   

         1996        1997        1998        1999        2000        2001  \
0  125.398693  124.772067  123.657335  122.560576  121.519841  120.546675   
1         NaN         NaN         NaN         NaN         NaN         NaN   
2         NaN         NaN         NaN         NaN         NaN         NaN   
3         NaN         NaN         NaN         NaN         NaN         NaN   
4         NaN         NaN         NaN         NaN         NaN         NaN   

         2002        2003        2004        2005        2006        2007  \
0 

In [ ]:
hnp_df['Indicator Name'].unique()

array(['Adolescent fertility rate (births per 1,000 women ages 15-19)',
       'Adults (ages 15+) and children (0-14 years) living with HIV',
       'Adults (ages 15+) and children (ages 0-14) newly infected with HIV',
       'Adults (ages 15+) living with HIV',
       'Adults (ages 15-49) newly infected with HIV',
       'Age at first marriage, female', 'Age at first marriage, male',
       'Age dependency ratio (% of working-age population)',
       'Age dependency ratio, old', 'Age dependency ratio, young',
       'Age population, age 00, female, interpolated',
       'Age population, age 00, male, interpolated',
       'Age population, age 01, female, interpolated',
       'Age population, age 01, male, interpolated',
       'Age population, age 02, female, interpolated',
       'Age population, age 02, male, interpolated',
       'Age population, age 03, female, interpolated',
       'Age population, age 03, male, interpolated',
       'Age population, age 04, female, interpolated

# Group 3
https://wandakinasih.my.canva.site/nbv1s

## ✍ Q1. Which country has the highest total fertility rate in 2019?

In [ ]:
# create fertility data frame
hfr = hnp_df[['Country Name', 'Country Code', 'Indicator Name', '2019']]
hfr = hfr[hfr['Indicator Name'] == 'Fertility rate, total (births per woman)']
hfr.head()

Country Name Country Code  \
119      Africa Eastern and Southern          AFE   
560       Africa Western and Central          AFW   
1001                      Arab World          ARB   
1442          Caribbean small states          CSS   
1883  Central Europe and the Baltics          CEB   

                                Indicator Name      2019  
119   Fertility rate, total (births per woman)  4.349433  
560   Fertility rate, total (births per woman)  5.044144  
1001  Fertility rate, total (births per woman)  3.187157  
1442  Fertility rate, total (births per woman)  1.979904  
1883  Fertility rate, total (births per woman)  1.555125

In [ ]:
# determine the highest total fertility rate in 2019 (using sort_values)
hfr.sort_values(by=['2019'], ascending = False).head(1)

Country Name Country Code                            Indicator Name  \
84791        Niger          NER  Fertility rate, total (births per woman)   

        2019  
84791  6.824

In [ ]:
# determine the highest total fertility rate in 2019 (using max)
hfr[hfr['2019'] == hfr['2019'].max()]

Country Name Country Code                            Indicator Name  \
84791        Niger          NER  Fertility rate, total (births per woman)   

        2019  
84791  6.824

### 📚 **Niger** has the highest total fertility rate of `6.824` total births per woman.

## ✍ Q2. Which country has the lowest rate of diabetes prevalence in 2021?

In [ ]:
# create data frame
ldp = hnp_df[['Country Name', 'Country Code', 'Indicator Name', '2021']]
ldp = ldp[ldp['Indicator Name'] == 'Diabetes prevalence (% of population ages 20 to 79)']
ldp.head()

Country Name Country Code  \
102      Africa Eastern and Southern          AFE   
543       Africa Western and Central          AFW   
984                       Arab World          ARB   
1425          Caribbean small states          CSS   
1866  Central Europe and the Baltics          CEB   

                                         Indicator Name       2021  
102   Diabetes prevalence (% of population ages 20 t...   7.388179  
543   Diabetes prevalence (% of population ages 20 t...   3.397262  
984   Diabetes prevalence (% of population ages 20 t...  14.033885  
1425  Diabetes prevalence (% of population ages 20 t...  11.770287  
1866  Diabetes prevalence (% of population ages 20 t...   6.634120

In [ ]:
# determine the lowest rate of diabetes prevelance in 2021 (using sort_values)
ldp.sort_values(by=['2021'], ascending = True).head(1)

Country Name Country Code  \
30531        Benin          BEN   

                                          Indicator Name  2021  
30531  Diabetes prevalence (% of population ages 20 t...   1.1

### 📚 **Benin** has the lowest rate of diabetes prevelance with total `1.1%` of population ages 20 to 79.

## ✍ Q3. Which country has the lowest prevalence of overweight adults in 2016?

In [ ]:
# create data frame
lpo = hnp_df[['Country Name', 'Country Code', 'Indicator Name', '2016']]
lpo = lpo[lpo['Indicator Name'] == 'Prevalence of overweight (% of adults)']
lpo.head()

Country Name Country Code  \
345      Africa Eastern and Southern          AFE   
786       Africa Western and Central          AFW   
1227                      Arab World          ARB   
1668          Caribbean small states          CSS   
2109  Central Europe and the Baltics          CEB   

                              Indicator Name       2016  
345   Prevalence of overweight (% of adults)  28.917314  
786   Prevalence of overweight (% of adults)  28.868763  
1227  Prevalence of overweight (% of adults)  61.843016  
1668  Prevalence of overweight (% of adults)  53.123588  
2109  Prevalence of overweight (% of adults)  59.044895

In [ ]:
# determine the lowest perevelance of overweight adults in 2016 (using sort_values)
lpo.sort_values(by=['2016'], ascending = True).head(1)

Country Name Country Code                          Indicator Name  2016
115005      Vietnam          VNM  Prevalence of overweight (% of adults)  18.3

### 📚 **Vietnam** has the lowest prevelance of overweight aduls in 2016 with probability of `18.3%` adults.

## ✍ Q4. How many Indonesian children (age <15) are living with HIV in 2020?

In [ ]:
# create data frame
hiv = hnp_df[['Country Name', 'Country Code', 'Indicator Name', '2020']]
hiv = hiv[hiv['Indicator Name'] == 'Children (0-14) living with HIV']
hiv.head()

Country Name Country Code  \
72       Africa Eastern and Southern          AFE   
513       Africa Western and Central          AFW   
954                       Arab World          ARB   
1395          Caribbean small states          CSS   
1836  Central Europe and the Baltics          CEB   

                       Indicator Name  2020  
72    Children (0-14) living with HIV   NaN  
513   Children (0-14) living with HIV   NaN  
954   Children (0-14) living with HIV   NaN  
1395  Children (0-14) living with HIV   NaN  
1836  Children (0-14) living with HIV   NaN

In [ ]:
# determine amount of Indonesian children with HIV in 2020
hiv[hiv['Country Name']=='Indonesia']

Country Name Country Code                   Indicator Name     2020
61371    Indonesia          IDN  Children (0-14) living with HIV  18000.0

### 📚 `18,000` children (age <15) in Indonesia living with HIV.

## ✍ Q5. Which country has the highest probability of dying per 1000 among adolescents ages 5-9 years in 2020?

In [ ]:
# create data frame
hpd = hnp_df[['Country Name', 'Country Code', 'Indicator Name', '2020']]
hpd = hpd[hpd['Indicator Name'] == 'Probability of dying among children ages 5-9 years (per 1,000)']
hpd.head()

Country Name Country Code  \
373      Africa Eastern and Southern          AFE   
814       Africa Western and Central          AFW   
1255                      Arab World          ARB   
1696          Caribbean small states          CSS   
2137  Central Europe and the Baltics          CEB   

                                         Indicator Name       2020  
373   Probability of dying among children ages 5-9 y...   7.992731  
814   Probability of dying among children ages 5-9 y...  12.572555  
1255  Probability of dying among children ages 5-9 y...   3.237229  
1696  Probability of dying among children ages 5-9 y...   1.301929  
2137  Probability of dying among children ages 5-9 y...   0.500265

In [ ]:
# determine country with highest probability of dying among adolecense age 5-9 in 2020
hpd.sort_values(by=['2020'], ascending = False).head(1)

Country Name Country Code  \
99598      Somalia          SOM   

                                          Indicator Name  2020  
99598  Probability of dying among children ages 5-9 y...  16.1

### 📚 **Somalia** has the highest probability of dying per 1000 among adolescents ages 5-9 years in 2020 with probability of `0,16%`.